carmirror/store.go

```diff
@@ -13,26 +15,38 @@ import (
 
 type KuboStore struct {
 	store ipld.DAGService
+	lng   ipld.NodeGetter
 	pins  kubo.PinAPI
 }
 
 func NewKuboStore(core kubo.CoreAPI) *KuboStore {
+	// TODO: pass errors back to caller instead of panicking
+	lng, err := NewLocalNodeGetter(core)
+	if err != nil {
+		panic(err)
+	}
+
 	return &KuboStore{
-		core.Dag(),
-		core.Pin(),
+		store: core.Dag(),
+		lng:   lng,
+		pins:  core.Pin(),
 	}
 }
 
 func (ks *KuboStore) Get(ctx context.Context, cid cmipld.Cid) (cm.Block[cmipld.Cid], error) {
-	if node, err := ks.store.Get(ctx, cid.Unwrap()); err != nil {
+	if node, err := ks.lng.Get(ctx, cid.Unwrap()); err != nil {
```

Comment:
Had to add the local node getter, or Get would hang and therefore the entire session would hang.